devlib/module/sched.py

```diff
@@ -268,6 +268,58 @@ def set_kernel_attribute(self, attr, value, verify=True):
         path = '/proc/sys/kernel/sched_' + attr
         self.target.write_value(path, value, verify)
 
+    @property
+    @memoized
+    def has_debug(self):
+        if self.target.config.get('SCHED_DEBUG') != 'y':
+            logger.warn("target's kernel compiled with !CONFIG_SCHED_DEBUG")
+        return self.target.file_exists('/sys/kernel/debug/sched_features')
+
+    def get_features(self):
+        """
+        Get the status of each sched feature
+
+        :returns: a dictionary of features and their "is enabled" status
+        """
+        if not self.has_debug:
+            raise RuntimeError("sched_features not available")
+        feats = self.target.read_value('/sys/kernel/debug/sched_features')
+        features = {}
+        for feat in feats.split():
+            value = True
+            if feat.startswith('NO'):
+                feat = feat.replace('NO_', '')
+                value = False
+            features[feat] = value
+        return features
+
+    def set_feature(self, feature, enable, verify=True):
+        """
+        Set the status of a specified scheduler feature
+
+        :param feature: the feature name to set
+        :param enable: true to enable the feature, false otherwise
+
+        :raise ValueError: if the specified enable value is not bool
+        :raise RuntimeError: if the specified feature cannot be set
+        """
+        if not self.has_debug:
+            raise RuntimeError("sched_features not available")
+        if not isinstance(enable, bool):
+            raise ValueError('enable parameter can only be boolean')
+        feat_value = feature.upper()
+        if enable is False:
```

Comment:
`if not enable:` would be more Pythonic

Reply:
yes... although sometimes I've got the impression that if we wanna ensure a specific value/object (i.e. a boolean false in this case) it's usually better to check it explicit. I'm always a bit confused on that.
But I guess here we are ensure by the before coerce to boolean that this `not` will mean `False`.

Reply:
Yes, given the coercion above, there is no practical difference. My comment was purely stylistic.